jscomp/ml/matching.ml

```diff
@@ -3072,15 +3002,7 @@ let for_let loc param pat body =
       (* fast path, and keep track of simple bindings to unboxable numbers *)
       Llet(Strict, Pgenval, id, param, body)
   | _ ->
-      (* Turn off such optimization to reduce diff in the beginning - FIXME*)
```

Comment:
An interesting optimisation, which might be worth resurrecting in the future 
![image](https://github.com/user-attachments/assets/2a091f2a-3b93-4442-9d35-de9e05ba3b12)